### PR TITLE
docs(dpg): update `ProtocolMethods.md`

### PR DESCRIPTION
**Pull request by archerzz**
- update `ProtocolMethods.md` to be more aligned with existing `PetsStoreClient` and `Pet`.
- remove section regarding `RequestContext` since it's a duplicate, and add links to overall readme and sample index.
- remove unused `id` property from `Pet` model
- add a code snippet directive in unit test, for sample contents
- fix parameter names in `PetStoreClient`

part of #26454

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by archerzz** [comment]
This PR focuses on cleaning up some obvious errors in the `ProtocolMethods.md`.

We still have a fundamental problem to solve: `PetStoreClient` is not generated, which will be resolved in #28315

File: sdk/core/Azure.Core/samples/ProtocolMethods.md [hunk 1]
```diff
@@ -4,23 +4,22 @@
 
 Azure SDK clients provide an interface to Azure services by translating library calls to REST requests.
 
-The schematized body in the request or response, known as the message body, can be exposed in the Azure SDK client one of two ways:
+In Azure SDK clients, there are two ways to expose the schematized body in the request or response, known as the `message body`:
 
-Most Azure SDK Clients expose methods that take ['model types'](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-model-types) as parameters, C# classes which map to the message body of the REST call. Those methods can be called here '**standard model methods**',
+- Most Azure SDK Clients expose methods that take ['model types'](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-model-types) as parameters, C# classes which map to the `message body` of the REST call. Those methods can be called here '**standard model methods**'.
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/samples/ProtocolMethods.md:9]:
```suggestion
- Most Azure SDK Clients expose methods that take ['model types'](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-model-types) as parameters, C# classes which map to the `message body` of the REST call. Those methods can be called here '**convenience methods**'.
```

File: sdk/core/Azure.Core/samples/ProtocolMethods.md [hunk 2]
```diff
@@ -82,65 +81,43 @@ using Azure.Identity;
...
-int code = response.Status;
+var doc = JsonDocument.Parse(response.Content.ToMemory());
+var name = doc.RootElement.GetProperty("name").GetString();
 ```
 
-### More on `RequestContext`
+## Configuration And Customization
 
-For more usage details, check [RequestContext samples](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/RequestContext.md)
+**Protocol methods** share the same configuration and customization as **standard model methods**. For details, see the [ReadMe](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md). You can find more samples [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/README.md).
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/samples/ProtocolMethods.md:123]:
```suggestion
**Protocol methods** share the same configuration and customization as **convenience methods**. For details, see the [ReadMe](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/README.md). You can find more samples [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/README.md).
```

**Review by annelo-msft — APPROVED** [review]
Please update "standard model methods" to "convenience methods."  Otherwise, this looks good at this time, and we may revisit details as part of https://github.com/azure/azure-sdk-for-net/issues/27770

**Pull request merged**